test.go

```diff
@@ -2,20 +2,21 @@ package main
 
 import (
 	"crypto/rand"
+	"fmt"
 	"math/big"
 )
 
-func Test(f FiniteField) bool {	
+func GFPTest(f GFP) bool {
 	a, _ := rand.Int(rand.Reader, f.prime)
```

Comment:
rand is deterministically seeded when running tests (for easy reproducibility). 
You might want to consider seeding it anew with  rand.Seed(time.Now().UnixNano())